src/biome/text/models/base_model_classifier.py

```diff
@@ -19,7 +18,141 @@
+        """
+        Does a simple position-wise argmax over each token, converts indices to string labels, and
+        adds a ``"tags"`` key to the dictionary with the result.
+        """
+        all_predictions = output_dict["class_probabilities"]
+        if not isinstance(all_predictions, np.ndarray):
+            all_predictions = all_predictions.data.numpy()
+
+        output_map_probs = []
+        max_classes = []
+        max_classes_prob = []
+        for i, probs in enumerate(all_predictions):
+            argmax_i = np.argmax(probs)
+            label = self.vocab.get_token_from_index(argmax_i, namespace="labels")
+            label_prob = 0.0
+
+            output_map_probs.append({})
+            for j, prob in enumerate(probs):
+                label_key = self.vocab.get_token_from_index(j, namespace="labels")
+                output_map_probs[i][label_key] = prob
+                if label_key == label:
+                    label_prob = prob
+
+            max_classes.append(label)
+            max_classes_prob.append(label_prob)
+
+        return_dict = {
+            "logits": output_dict.get("logits"),
+            "classes": output_map_probs,
+            "max_class": max_classes,
+            "max_class_prob": max_classes_prob,
+        }
+        # having loss == None in dict (when no label is present) fails
+        if "loss" in output_dict.keys():
+            return_dict["loss"] = output_dict.get("loss")
+        return return_dict
+
+    def get_metrics(self, reset: bool = False) -> Dict[str, float]:
+        """Get the metrics of our classifier, see :func:`~allennlp_2.models.Model.get_metrics`.
+
+        Parameters
+        ----------
+        reset
+            Reset the metrics after obtaining them?
+
+        Returns
+        -------
+        A dictionary with all metric names and values.
+        """
+        all_metrics = {}
+
+        total_f1 = 0.0
+        total_precision = 0.0
+        total_recall = 0.0
+        for metric_name, metric in self._biome_classifier_metrics.items():
+            if metric_name == "accuracy":
+                all_metrics["accuracy"] = metric.get_metric(reset)
+            else:
+                precision, recall, f1 = metric.get_metric(
+                    reset
+                )  # pylint: disable=invalid-name
+                total_f1 += f1
+                total_precision += precision
+                total_recall += recall
+                all_metrics[metric_name + "/f1"] = f1
+                all_metrics[metric_name + "/precision"] = precision
+                all_metrics[metric_name + "/recall"] = recall
+
+        num_metrics = len(self._biome_classifier_metrics)
+        all_metrics["average/f1"] = total_f1 / num_metrics
+        all_metrics["average/precision"] = total_precision / num_metrics
+        all_metrics["average/recall"] = total_recall / num_metrics
+
+        return all_metrics
+
+
+class BaseModelClassifier(BiomeClassifierMixin, Model):
```

Comment:
Now, the BaseModelClassifier class is not a proper base class, but keep shared code for our SequenceClassifier and SequencePairClassifier implementations for the architecture initialization. The only value is changing is the number of input. 1 for sequence classifier and 2 for sequence pair classifier. 

It should be nice redefine those architectures for manage with one single class, avoiding this messy heritage 

Anyway, the naming used is not the best right now. We can call it `SequenceClassifierBase` or `BaseSequenceClassifier`, while we think how to consolidate sequence and sequence-pair in a more elegant way